dts/bindings/adc/nordic,nrf-saadc.yaml

```diff
@@ -14,6 +14,16 @@ properties:
   interrupts:
     required: true
 
+  enable-internal-timer:
+    type: boolean
+    description: |
+      Enable the internal timer for the SAADC to improve interval accuracy.
+      When enabled, the SAADC will utilize its built-in local timer for sampling.
+      If disabled, the SAADC will rely on the kernel timer.
+      WARNING: The internal timer is only applicable for sequences with a single channel.
+               It can be used only for intervals between consecutive samplings ranging
+               from 500 ns to 128 us.
+
```

Comment:
This really should be handled internally in the driver, if a user request a sequence with only 1 channel, and the interval is within what the internal timer supports, use the internal timer. If a user requests a sequence with multiple channels, or unsupported interval, use kernel timer. A LOG_INF() can be added to inform the user which timer was used to let users know why the intervals may be less accurate than expected.